### PR TITLE
Double the number of exporter build-and-test groups

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** Double the number of exporter test groups so that new tests can be added, e.g., the new tests in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31996.

**Testing:** These changes were applied in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31996 and they resolved a timeout.

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -29,9 +29,11 @@ RECEIVER_MODS_2 := $(shell find ./receiver/[p]* $(FIND_MOD_ARGS) -exec $(TO_MOD_
 RECEIVER_MODS_3 := $(shell find ./receiver/[q-z]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
 RECEIVER_MODS := $(RECEIVER_MODS_0) $(RECEIVER_MODS_1) $(RECEIVER_MODS_2) $(RECEIVER_MODS_3)
 PROCESSOR_MODS := $(shell find ./processor/* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-EXPORTER_MODS_0 := $(shell find ./exporter/[a-m]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-EXPORTER_MODS_1 := $(shell find ./exporter/[n-z]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
-EXPORTER_MODS := $(EXPORTER_MODS_0) $(EXPORTER_MODS_1)
+EXPORTER_MODS_0 := $(shell find ./exporter/[a-c]* $(FIND_MOD_ARGS) -exec $(TO_MOD_DIR) )
```
Comment (crobert-1) [inline Makefile:32]: Just a note for context, this split ends up being the following number of exporters in each group:

Group 0: 13
Group 1: 10
Group 2: 8
Group 3: 12

<details>
<summary>Here's the full split, if anyone's interested </summary>
<br>

 ```
./exporter/alertmanagerexporter
./exporter/alibabacloudlogserviceexporter
./exporter/awscloudwatchlogsexporter
./exporter/awsemfexporter
./exporter/awskinesisexporter
./exporter/awss3exporter
./exporter/awsxrayexporter
./exporter/azuredataexplorerexporter
./exporter/azuremonitorexporter
./exporter/carbonexporter
./exporter/clickhouseexporter
./exporter/cassandraexporter
./exporter/coralogixexporter

./exporter/datadogexporter
./exporter/datasetexporter
./exporter/elasticsearchexporter
./exporter/fileexporter
./exporter/googlecloudexporter
./exporter/googlecloudpubsubexporter
./exporter/googlemanagedprometheusexporter
./exporter/honeycombmarkerexporter
./exporter/influxdbexporter
./exporter/instanaexporter

./exporter/kafkaexporter
./exporter/loadbalancingexporter
./exporter/logicmonitorexporter
./exporter/logzioexporter
./exporter/lokiexporter
./exporter/mezmoexporter
./exporter/opencensusexporter
./exporter/opensearchexporter

./exporter/prometheusexporter
./exporter/prometheusremotewriteexporter
./exporter/pulsarexporter
./exporter/sapmexporter
./exporter/sentryexporter
./exporter/signalfxexporter
./exporter/skywalkingexporter
./exporter/splunkhecexporter
./exporter/sumologicexporter
./exporter/syslogexporter
./exporter/tencentcloudlogserviceexporter
./exporter/zipkinexporter
```

</details>

Given how uneven the distribution is in component names, I think this is as good of a split as possible with the current set of exporters. 👍

**Review by crobert-1 — APPROVED** [review]

**Comment by crobert-1** [comment]
_Failing test is frequency of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27295, I've added it as frequency there._

**Review by songy23 — COMMENTED** [review]
Can you update `.github/workflows/build-and-test-windows.yml` as well?

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32510 for context. cc @mx-psi

**Review by dashpole — APPROVED** [review]

**Review by crobert-1 — CHANGES_REQUESTED** [review]
I just realized you'll also need to update `build-and-test-windows.yml` with the new exporter groups.

**Comment by jmacd** [comment]
Oh! The windows build-and-test appears out-of-date prior to my change--apparently was skipping some receiver tests.
See [feb5ba2](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32496/commits/feb5ba249ca134273b38482b76e06d4093c13b79)

**Comment by crobert-1** [comment]
> Oh! The windows build-and-test appears out-of-date prior to my change--apparently was skipping some receiver tests. See [feb5ba2](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32496/commits/feb5ba249ca134273b38482b76e06d4093c13b79)

You're right, we're fixing it in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32518, which is what made me remember this would need updated too. 👍 

If possible, maybe only add the new groups to the Windows tests, and let the other PR add the rest? That way the changelog will be a bit more clear going forward.

**Review by crobert-1 — APPROVED** [review]
Thanks for making the extra changes!

**Comment by jmacd** [comment]
```
=== Errors
multiple definitions of TestMain
```

**Comment by crobert-1** [comment]
> ```
> === Errors
> multiple definitions of TestMain
> ```

Should be fixed by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32603. Hopefully another rebase should fix it.

**Pull request merged**